tests/FileHelperTest.php

```diff
@@ -220,16 +272,501 @@ public function testNormalizePath(): void
+            'data' => [],
+            'data-backup' => [],
+        ]);
+
+        FileHelper::copyDirectory(
+            $this->testFilePath . '/data',
+            $this->testFilePath . '/data-backup'
+        );
+
+        $this->assertTrue(true, 'no error');
+    }
+
+    public function testsCopyDirectoryFilterPath()
+    {
+        $source = 'boostrap4';
+
+        $structure = [
+            'css' => [
+                'bootstrap.css'           => 'file 1 content',
+                'bootstrap.css.map'       => 'file 2 content',
+                'bootstrap.min.css'       => 'file 3 content',
+                'bootstrap.min.css.map'   => 'file 4 content'
+            ],
+            'js' => [
+                'bootstrap.js'            => 'file 5 content',
+                'bootstrap.bundle.js'     => 'file 6 content',
+                'bootstrap.bundle.js.map' => 'file 7 content',
+                'bootstrap.min.js'        => 'file 8 content'
+            ]
+        ];
+
+        $this->createFileStructure([
+            $source => $structure,
+        ]);
+
+        $basePath = $this->testFilePath;
+        $source = $basePath . '/' . $source;
+        $destination = $basePath . '/assets';
+
+        // without filter options return all directory.
+        $options = [];
+
+        FileHelper::copyDirectory($source, $destination, $options);
+
+        $this->assertFileExists($destination, 'Destination directory does not exist!');
+
+        $checker = function ($structure, $dstDirName) use (&$checker) {
+            foreach ($structure as $name => $content) {
+                if (is_array($content)) {
+                    $checker($content, $dstDirName . '/' . $name);
+                } else {
+                    $fileName = $dstDirName . '/' . $name;
+                    $this->assertFileExists($fileName);
+                    $this->assertStringEqualsFile($fileName, $content, 'Incorrect file content!');
+                }
+            }
+        };
+
+        $checker($structure, $destination);
+    }
+
+    public function testsCopyDirectoryFilterPathOnly()
+    {
+        $source = 'boostrap4';
+
+        $structure = [
+            'css' => [
+                'bootstrap.css'           => 'file 1 content',
+                'bootstrap.css.map'       => 'file 2 content',
+                'bootstrap.min.css'       => 'file 3 content',
+                'bootstrap.min.css.map'   => 'file 4 content'
+            ],
+            'js' => [
+                'bootstrap.js'            => 'file 5 content',
+                'bootstrap.bundle.js'     => 'file 6 content',
+                'bootstrap.bundle.js.map' => 'file 7 content',
+                'bootstrap.min.js'        => 'file 8 content'
+            ]
+        ];
+
+        $exist = [
+            'css' => [
+                'bootstrap.css'           => 'file 1 content',
+                'bootstrap.min.css'       => 'file 3 content',
+            ]
+        ];
+
+        $noexist = [
```

Comment:
`$noexist` is not used.

Reply:
If it is used, https://github.com/terabytesoftw/files/blob/6d4e1064f4d3a587b228e8dd0c99094f7abc237d/tests/FileHelperTest.php#L656

Reply:
It is not. It is overwritten there but not used.

Reply:
I already solved it create two functions and delete the duplicate code you can verify that scrutinizer no longer shows the error